src/enclave/claims.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "crypto/hash_provider.h"
+#include "ds/logger.h"
+
+namespace ccf
+{
+  class ClaimsDigest
+  {
+  public:
+    using Digest = crypto::Sha256Hash;
+
+  private:
+    bool is_set = false;
+    Digest digest;
+
+  public:
+    ClaimsDigest() = default;
+
+    inline void set(Digest&& digest_)
+    {
+      is_set = true;
+      digest = std::move(digest_);
+    }
+
+    inline void set(Digest::Representation&& r)
+    {
+      is_set = true;
+      digest.set(std::move(r));
+    }
+
+    inline bool empty() const
+    {
+      return !is_set;
+    }
+
+    const Digest& value() const
+    {
+      return digest;
+    }
+
+    bool operator==(const ClaimsDigest& other) const
+    {
+      return (is_set == other.is_set) && (digest == other.digest);
+    }
+  };
+
+  static ClaimsDigest no_claims()
+  {
+    return ClaimsDigest();
+  }
+
+  static crypto::Sha256Hash entry_leaf(
+    const std::vector<uint8_t> write_set,
+    const crypto::Sha256Hash& claims_digest)
+  {
+    crypto::Sha256Hash write_set_digest({write_set.data(), write_set.size()});
```

Comment:
```suggestion
    ClaimsDigest::Digest write_set_digest({write_set.data(), write_set.size()});
```

Here and in other places in this file.